scc/cert/serialization/types.go

```diff
@@ -0,0 +1,77 @@
+package serialization
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type HexBytes []byte
+
+func (h *HexBytes) UnmarshalJSON(data []byte) error {
+	var s string
+	err := json.Unmarshal(data, &s)
+	if err != nil {
+		return err
+	}
+	if s == "null" {
+		*h = nil
+		return nil
+	}
+	if !strings.HasPrefix(s, "0x") {
+		return fmt.Errorf("invalid hex string %s", s)
+	}
+	s = strings.TrimPrefix(s, "0x")
+	if len(s)%2 == 1 {
+		s = "0" + s
+	}
+	raw, err := hex.DecodeString(s)
+	if err != nil {
+		return err
+	}
+	*h = raw
+	return nil
+}
+
+func UnmarshallFixLenghtHexBytes(data []byte, length int) (HexBytes, error) {
+	var h HexBytes
+	err := h.UnmarshalJSON(data)
+	if err != nil {
+		return nil, err
+	}
+	if len(h) != length {
+		return nil, fmt.Errorf("invalid length %d, expected %d", len(h), length)
+	}
+	return h, nil
+}
+
+type PublicKey [48]byte
```

Comment:
There is nothing specific about this type that restricts its use to public keys. Anything that represents 48-byte could use this utility type for JSON serialization. I would thus suggest to use a more generic name like `HexBytes48`; same for the `Signature` below.

Reply:
using `HexBytes48` and `HexBytes96`